specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/preview/2017-04-01-preview/subscriptions.json

```diff
@@ -69,6 +65,110 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.StreamAnalytics/locations/{location}/testQuery": {
+      "post": {
+        "tags": [
+          "Subscriptions"
+        ],
+        "operationId": "Subscriptions_TestQuery",
+        "description": "Test the Stream Analytics query before submitting the job.",
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "List subscription quota information in West US": {
+            "$ref": "./examples/Subscription_TestQuery.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../common/v1/definitions.json#/parameters/LocationParameter"
+          },
+          {
+            "$ref": "../../common/v1/definitions.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../common/v1/definitions.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "The request to test the Stream Analytics query was successfully initiated."
+          },
+          "400": {
+            "description": "The request to test the Stream Analytics query was not formed correctly.",
+            "schema": {
+              "$ref": "../../common/v1/definitions.json#/definitions/Error"
+            }
+          },
+          "429": {
+            "description": "The subscription has submitted too many query-testing requests in a short time frame.",
+            "schema": {
+              "$ref": "../../common/v1/definitions.json#/definitions/Error"
+            }
+          },
+          "default": {
+            "description": "Error.",
+            "schema": {
+              "$ref": "../../common/v1/definitions.json#/definitions/Error"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.StreamAnalytics/locations/{location}/operationResults/{operationId}": {
+      "get": {
+        "tags": [
+          "Subscriptions"
+        ],
+        "operationId": "Subscriptions_GetOperationResults",
+        "description": "Fetch the asynchronous operation result.",
+        "x-ms-long-running-operation": true,
```

Comment:
Don't think GET operation results is an async API/long running operation. The operation it is tracking may be async/long running but the act of retrieving the result is not async/long running.

Also, not sure if we even need to define this API. I don't see it defined for other async APIs that we have. Doesn't seem super useful to expose as customers wouldn't have a way to know which operation results exists to be able to retrieve them, they usually just use whatever url is given in the location header.

Maybe this is something swagger/ARM can answer/give guidance.

Reply:
+1. This api should not be async and also you don't need to define it in the swagger.

---
In reply to: [591065672](https://github.com/Azure/azure-rest-api-specs/pull/13322#discussion_r591065672) [](ancestors = 591065672)